### PR TITLE
Add info about the checksum's size field

### PR DESCRIPTION
**Pull request by herbygillot**
So that poor @pmetzger does not have to repeat himself for every new ports submission.

### CODE REVIEW

**Comment by pmetzger** [comment]
Excellent! We should also add something to `port lint` to check for `size` fields, but that can happen after this gets merged. This LGTM, hopefully others agree.

File: guide/xml/portfile-phase.xml [hunk 1]
```diff
@@ -1044,7 +1044,8 @@ svn.revision        2100</programlisting>
         <listitem>
           <para>Checksum(s) of the distribution files. For ports with multiple
           distribution files, filenames must be included to associate files
-          with their checksums.</para>
+          with their checksums.  Each checksum entry must also indicate the
+          file's size.</para>
```
Comment (ryandesign) [inline guide/xml/portfile-phase.xml]: I would go with "should" here, as below, instead of "must".

**Comment by ryandesign** [comment]
> We should also add something to `port lint` to check for `size` fields

See also https://trac.macports.org/ticket/37667

File: guide/xml/portfile-phase.xml (hunk 1 above)
Comment (herbygillot) [reply guide/xml/portfile-phase.xml]: Fixed.

**Comment by herbygillot** [comment]
So out of curiosity, would a missing `size` field in the checksums section be a `lint` warning, or full-on error?

**Comment by pmetzger** [comment]
> So out of curiosity, would a missing `size` field in the checksums section be a lint warning, or full-on `error`?

I'm not sure what proper practice is, and both the guide and the man page are silent about what the distinction is. However, regardless, even warning is certainly better than what we have now.

**Comment by herbygillot** [comment]
I've made an attempt at a lint check in macports/macports-base#101
I will look at adding the digest checks as per the Trac ticket @ryandesign referenced perhaps after the above makes it in.

**Comment by jmroot** [comment]
> So out of curiosity, would a missing size field in the checksums section be a lint warning, or full-on error?

It shouldn't be either IMO, since omitting it causes no problem whatsoever. OTOH, similar to the ticket that Ryan linked, using _only_ a size should cause lint to complain.

**Comment by pmetzger** [comment]
@jmroot No. If we demand that people include `size`, and when I don't ask that I get yelled at, then it's a lint problem.

**Comment by jmroot** [comment]
People shouldn't be yelling at you for that, and nobody should be demanding that it be included. _It currently adds nothing._ Making it a barrier to contributing is ridiculous.

**Comment by l2dy** [comment]
We recommend adding `size` to checksums (https://github.com/macports/macports-base/commit/95a153e6a4bff8b99502806960c916a2b5d20504). IMHO, we shouldn't remove default checksums if it already exists in a Portfile (https://github.com/macports/macports-ports/pull/2371/commits/869606dc8fd4ca29aefc1946f08871420b9b7b46#diff-8c1882979658df3c168c7b489e3ae5d5L28).

**Comment by pmetzger** [comment]
@jmroot Look, I'm not willing to argue with you about this. I didn't want it. The current thought is that it is supposed to be used in an upcoming patch so that download progress bars work correctly even for sites like Github that don't provide length information before the fetch begins, and given that, it needs to be universal. If you don't like it, bring it up on the -dev list.

**Comment by ryandesign** [comment]
> would a missing size field in the checksums section be a lint warning, or full-on error?

We want portfiles to list the size in checksums, so not having it should be a lint warning, but not having a size doesn't currently cause any problems, and even in the future it would merely mean less-detailed progress information, so lack of size should not be considered an error.

A PR shouldn't be rejected just because size is missing from checksums. However, if people are submitting portfiles without size in checksums, we should figure out why, and fix it. One reason may be lack of mention in the guide, so this PR adding that info to the guide is great.

**Pull request merged**